merkle/smt/hasher.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package smt
+
+import (
+	"github.com/google/trillian/merkle/hashers"
+	"github.com/google/trillian/storage/tree"
+)
+
+// mapHasher is a wrapper around MapHasher bound to a specific tree ID.
+type mapHasher struct {
+	mh     hashers.MapHasher
+	treeID int64
+}
+
+// wrapHasher returns a mapHasher binding the given hasher to a tree ID.
+func wrapHasher(hasher hashers.MapHasher, treeID int64) mapHasher {
```

Comment:
Maybe `bindHasher`?

Reply:
I think `bindHasher` would be clearer. `forTree` etc. doesn't quite work as you pass the hasher in.